atef/qt_helpers.py

```diff
@@ -271,3 +276,53 @@ def put_to_index(self, index: int, new_value: Any) -> None:
         self.changed_value.emit(new_value)
         self.changed_index.emit(index)
         self.updated.emit()
+
+
+class ThreadWorker(QtCore.QThread):
```

Comment:
Good utility for long-running functions, I like it. We can connect the return signal to an appropriate slot and we'll not freeze the gui to do something intensive.